stubs/setuptools/setuptools/msvc.pyi

```diff
@@ -0,0 +1,169 @@
+import sys
+from typing import ClassVar, Final, TypedDict, overload, type_check_only
+from typing_extensions import LiteralString, NotRequired
+
+class winreg:
+    if sys.platform == "win32":
+        HKEY_USERS: Final[int]
+        HKEY_CURRENT_USER: Final[int]
+        HKEY_LOCAL_MACHINE: Final[int]
+        HKEY_CLASSES_ROOT: Final[int]
+    else:
+        HKEY_USERS: Final[None]
+        HKEY_CURRENT_USER: Final[None]
+        HKEY_LOCAL_MACHINE: Final[None]
+        HKEY_CLASSES_ROOT: Final[None]
+
+environ: dict[str, str]
```

Comment:
If we expose this internal implementation mock, I'd prefer re-exporting the real type on windows:
```suggestion
if sys.platform == "win32":
    import winreg as winreg 
    from os import environ as environ
else:
    class winreg:
        HKEY_USERS: Final[None]
        HKEY_CURRENT_USER: Final[None]
        HKEY_LOCAL_MACHINE: Final[None]
        HKEY_CLASSES_ROOT: Final[None]

   environ: dict[str, str]
```
It's also still present in https://github.com/python/typeshed/blob/main/stubs/setuptools/%40tests/stubtest_allowlist_darwin.txt , that entry or these mocks should be removed